catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -193,19 +193,17 @@ public Metacard getMetacard(String id)
         }
 
         Filter queryFilter = filterBuilder.anyOf(filters);
-        QueryResponse response = catalogFramework.query(new QueryRequestImpl(new QueryImpl(
-                queryFilter,
-                1,
-                -1,
-                SortBy.NATURAL_ORDER,
-                false,
-                TimeUnit.SECONDS.toMillis(10)), false));
-        Map<String, Result> results = new HashMap<>();
-        for (Result result : response.getResults()) {
-            results.put(result.getMetacard()
-                    .getId(), result);
-        }
-        return results;
+        ResultIterable resultIterable = new ResultIterable(catalogFramework,
+                new QueryRequestImpl(new QueryImpl(queryFilter,
+                        1,
+                        pageSize,
+                        SortBy.NATURAL_ORDER,
+                        false,
+                        TimeUnit.SECONDS.toMillis(10)), false));
+
+        return resultIterable.stream()
+                .collect(Collectors.toMap(result -> result.getMetacard()
+                        .getId(), result -> result));
```

Comment:
✏️  should be able to change this one to `Function.identity()` like the one above that chris mentioned